File: WINGS_AI_Bootcamp/Haskell/diamond.hs
import Data.List (intercalate)

line :: Int -> String
line k = intercalate " " (replicate (2 * k + 1) "*")

makeDiamond :: Int -> String
makeDiamond size =
    let mid = size `div` 2
        top = [line i | i <- [0..mid]]
        bottom = [line i | i <- reverse [0..(mid - 1)]]
        allLines = top ++ bottom
        width = maximum (map length allLines)
    in unlines [center width l | l <- allLines]

center :: Int -> String -> String
center w s =
    let pad = replicate ((w - length s) `div` 2) ' '
    in pad ++ s

main :: IO ()
main = putStrLn $ makeDiamond 7

File: WINGS_AI_Bootcamp/Python/diamond.py
def line(k: int) -> str:
    return ' '.join(['*'] * (2 * k + 1))

def make_diamond(size: int) -> str:
    mid = size // 2
    top = [line(i) for i in range(mid + 1)]
    bottom = [line(i) for i in reversed(range(mid))]
    all_lines = top + bottom
    width = max(len(l) for l in all_lines)
    return '\n'.join(l.center(width) for l in all_lines)

# Example usage
print(make_diamond(7))